README.md
tinyMCE 4 plugin qrcode
======================

A QRcode wizzard plugin for tinyMCE 4 ( Build with Bootstrap version 3 build 2 )

### installation

Create map qrcode in tinymce/plugins/

Upzip all files in to the map qrcode

Now add qrcode to the plugins: and toolbar:

```html
<script type="text/javascript">
tinymce.init({
	selector: "textarea",
	plugins: [
			"advlist autolink lists link image charmap print preview anchor",
			"searchreplace visualblocks code fullscreen",
			"insertdatetime media table contextmenu paste qrcode"
			],
	toolbar: "insertfile undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image qrcode"
	});
</script>
```

### Demo

For a demo <a href="http://www.cfcms.nl/tinymce-youtube/index.html#demo">see this page</a> or have a look at <a href="http://www.youtube.com/watch?v=Kps3Q081nQc">this video</a>

Also don't forget to get my other tinyMCE 4 plugin 'TinyMCE 4 plugin Youtube search and insert'

Grab it <a href="http://codecanyon.net/item/tinymce-4-plugin-youtube-search-and-insert/5495575?ref=ceasar">here</a>

Or my other tinyMCE 4 plugin 'TinyMCE 4 plugin Flickr image search and place'

Grab that one <a href="http://codecanyon.net/item/tinymce-4-plugin-flickr-image-search-and-place/5703888?ref=ceasar">here</a>

### Commercial use

For <b>commercial use</b> you can buy an unlimited license for one product for 25,-- EURO (excl. VAT) (for example a cms)

<a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=UP9CAH952MAFA">Click here to buy the license with paypal !</a>

## Author
**Ceasar Feijen (cfconsultancy)**
[http://www.cfconsultancy.nl](http://www.cfconsultancy.nl)

changelog.txt
Version 1.1.0 (2013-10-14)
	Changed API from google (deprecated) to api.qrserver.com V1
	Added bootstrap colorbox to produce colored images  

Version 1.0.0 (2013-09-15)
	First release.

js/qr.js
    var text_max = 300;
    $('#textarea_feedback').html(text_max + ' characters');

    $('#codeData').keyup(function() {
        var text_length = $('#codeData').val().length;
        var text_remaining = text_max - text_length;

        $('#textarea_feedback').html(text_remaining + ' characters');
    });

	$("#generate").click(function() {

	    var datainput = $("#codeData").val().replace(/\s/g, '+');
	    var size = $("#codeSize").val();
	    var encoding = $("#SelectCodding").val();
        var codecolor = $("#codeColor").val(function(i, v) {return v.replace("#","");}).val();


	    if(datainput == "") {
	        showalert('Please enter a url or text or look at the Help data formats !','alert-danger');
	        $("#generate").prop("disabled", true);
	        return false;

	    } else {

	        if( $("#image").is(':empty')) {
	            $("#image").append("<img src='http://api.qrserver.com/v1/create-qr-code/?data=" + encodeURIComponent(datainput) + "&qzone=2&color=" + codecolor + "&size=" + size + "&charset-target=" + encoding + "&ecc=L&format=png' data-name='" + datainput + "' />");
	            $("#arrow").show();
	            return false;
	        } else {
	            $("#image").html("");
	            $("#image").append("<img src='http://api.qrserver.com/v1/create-qr-code/?data=" + encodeURIComponent(datainput) + "&qzone=2&color=" + codecolor + "&size=" + size + "&charset-target=" + encoding + "&ecc=L&format=png' data-name='" + datainput + "' />");
	            $("#arrow").show();
	            return false;
	        }
	    }

	});

	$("#image").click(function() {

        //alert($("#image").html());
        //return false;
	    I_InsertHTML($("#image").html());
	    I_Close();
	});

	function I_InsertHTML(sHTML) {
	    parent.tinymce.activeEditor.insertContent(sHTML);
	};

	function I_Close() {
	    parent.tinymce.activeEditor.windowManager.close();
	};

    /**
      Bootstrap Alerts -
      Function Name - showalert()
      Inputs - message,alerttype
      Example - showalert("Invalid Login","alert-error")
      Types of alerts -- "alert-error","alert-success","alert-info"
      Required - You only need to add a alert_placeholder div in your html page wherever you want to display these alerts "<div id="alert_placeholder"></div>"
    **/

    function showalert(message,alerttype) {

      $('#alert_placeholder').append('<div id="alertdiv" class="alert ' +  alerttype + '"><a class="close" data-dismiss="alert">x</a><span>'+message+'</span></div>');

	  $('#alert_placeholder').bind('closed.bs.alert', function () {
	  	$("#generate").prop("disabled", false);
	  })

		setTimeout(function() { // this will automatically close the alert and remove this if the users doesnt close it in 5 secs

			$("#alertdiv").remove();
			$("#generate").prop("disabled", false);

		}, 3000);

	};

	$('#formatOption').change(function(){
	    var selected_item = $(this).val();
	    $('#codeData').val(selected_item);
	});

plugin.js
/**
 * QR code - a TinyMCE 4 QR code wizzard
 * qrcode/plugin.js
 *
 *
 * Plugin info: http://www.cfconsultancy.nl/
 * Author: Ceasar Feijen
 *
 * Version: 1.0 released 15/09/2013
 */

tinymce.PluginManager.add('qrcode', function(editor) {

    function openmanager() {
        var title="Create QRcode";
        if (typeof editor.settings.qrcode_title !== "undefined" && editor.settings.qrcode_title) {
            title=editor.settingsqrcode_title;
        }
        win = editor.windowManager.open({
            title: title,
            file: tinyMCE.baseURL + '/plugins/qrcode/qrcode.html',
            filetype: 'image',
	    	width: 550,
            height: 540,
            inline: 1,
            buttons: [{
                text: 'cancel',
                onclick: function() {
                    this.parent()
                        .parent()
                        .close();
                }
            }]
        });

    }
	editor.addButton('qrcode', {
		icon: true,
		image: tinyMCE.baseURL + '/plugins/qrcode/icon.png',
		tooltip: 'Create QRcode',
		shortcut: 'Ctrl+QR',
		onclick: openmanager
	});

	editor.addShortcut('Ctrl+QR', '', openmanager);

	editor.addMenuItem('qrcode', {
		icon:'media',
		text: 'Create QRcode',
		shortcut: 'Ctrl+QR',
		onclick: openmanager,
		context: 'insert'
	});
});
